src/config/graph.rs

```diff
@@ -176,10 +176,10 @@ impl Graph {
             let from_ty = self.get_output_type(&edge.from);
             let to_ty = self.get_input_type(&edge.to);
 
-            if from_ty != to_ty && from_ty != DataType::Any && to_ty != DataType::Any {
+            if !from_ty.intersects(to_ty) {
                 errors.push(format!(
-                    "Data type mismatch between {} ({:?}) and {} ({:?})",
-                    edge.from, from_ty, edge.to, to_ty
+                    "Data type mismatch between {} and {}",
+                    edge.from, edge.to
```

Comment:
Is there not a good way to keep the types in this error message? I get that it's a bit more complex now, but it does seem like helpful information that'd be nice to keep.